CMakeLists.txt

```diff
@@ -429,7 +429,7 @@ else()
 endif()
 
 # for development purposes, compile the exmaples together with everything else
-if (DEFINED Tasmanian_DEVELOPMENT_EXAMPLES AND Tasmanian_DEVELOPMENT_EXAMPLES)
+if (DEFINED Tasmanian_ENABLE_DEVELOPMENT_DEFAULTS AND Tasmanian_ENABLE_DEVELOPMENT_DEFAULTS)
```

Comment:
`if (Tasmanian_ENABLE_DEVELOPMENT_DEFAULTS)` already does what you want. It evaluate to false if undefined.

Reply:
Also I am getting confused by the body of that if-statement. Is the purpose of that variable only to enable the examples? If so all it should do is add the executable and link. Or even better, set a variable `Tasmanian_ENABLE_EXAMPLES` to true and later conditionally add the directory `Examples/` and let it have it's own `CMakeLists.txt` that adds the executable, place the headers in include search path and link libraries as needed.

Reply:
Examples have their own CMakeLists, which points to the installed version of Tasmanian. However, when I develop the examples, I need to constantly recompile and I didn't want to go through the sequence "CMake Tasmanian -> Install -> CMake Examples", I wanted to have the examples compiled in the Tasmanian build folder. This is nothing but convenience, and it does not affect the user in any way.
I used to have "Tasmanian_DEVELOPMENT_EXAMPLES", but I changed that to "Tasmanian_ENABLE_DEVELOPMENT_DEFAULTS" which covers both the examples and the CXX flags, the development section enables -Wall -Wextra -pedantic and so on. It is something for you and me to use, but no the user.

Reply:
The cmake system needs an overhaul, but I have an ECP milestone due at the end of the month and I have to make a large Tasmanian presentation in April. I will go over cmake once more, break it into sub-files for the different sections, and change all of the Defined statements. Right now, the statement above was changes with "search-replace".